DependencyInjection/HWIOAuthExtension.php

```diff
@@ -108,17 +108,27 @@ public function createResourceOwnerService(ContainerBuilder $container, $name, a
         if (isset($options['service'])) {
             // set the appropriate name for aliased services, compiler pass depends on it
             $container->setAlias('hwi_oauth.resource_owner.'.$name, $options['service']);
-        } else {
-            $type = $options['type'];
-            unset($options['type']);
 
+            return;
+        }
+
+        $type = $options['type'];
+        unset($options['type']);
+
+        // handle external resource owners with given class
+        if (isset($options['class'])) {
```

Comment:
check using `is_subclass_of` function [reference](http://php.net/manual/en/function.is-subclass-of.php)

```php
if (!is_subclass_of($options['class'], 'HWI\Bundle\OAuthBundle\OAuth\ResourceOwnerInterface')) {
    throw new InvalidConfigurationException(sprintf('Class "%s" must implement interface "HWI\Bundle\OAuthBundle\OAuth\ResourceOwnerInterface".', $options['class']));
}
```